File: data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/task/UserCacheUpdaterConfig.scala
package org.sunbird.dp.usercache.task

import com.typesafe.config.Config
import org.apache.flink.api.common.typeinfo.TypeInformation
import org.apache.flink.api.java.typeutils.TypeExtractor
import org.sunbird.dp.core.job.BaseJobConfig
import org.sunbird.dp.usercache.domain.Event

import java.util.{List => JList}

class UserCacheUpdaterConfig(override val config: Config) extends BaseJobConfig(config, "UserCacheUpdaterJob") {

  private val serialVersionUID = 2905979434303791379L

  implicit val mapTypeInfo: TypeInformation[Event] = TypeExtractor.getForClass(classOf[Event])

  // Kafka Topics Configuration
  val inputTopic: String = config.getString("kafka.input.topic")
  val userFields = List("usertype", "grade", "language", "subject", "state", "district", "usersignintype", "userlogintype","locationids")

  // User cache updater job metrics
  val userCacheHit = "user-cache-hit"
  val skipCount = "skipped-message-count"
  val successCount = "success-message-count"
  val dbReadSuccessCount = "db-read-success-count"
  val dbReadMissCount = "db-read-miss-count"
  val totalEventsCount ="total-audit-events-count"

  val userSelfSignedInTypeList: JList[String] = config.getStringList("user.self.signin.types")
  val userValidatedTypeList: JList[String] = config.getStringList("user.validated.types")
  val userSelfSignedKey: String = config.getString("user.self.signin.key")
  val userValidatedKey: String = config.getString("user.valid.key")

  // Redis
  val userStore: Int = config.getInt("redis-meta.database.userstore.id")

  // lms-cassandra
  val keySpace: String = config.getString("lms-cassandra.keyspace")
  val locationTable: String = config.getString("lms-cassandra.table.location")
  val userTable: String = config.getString("lms-cassandra.table.user")
  val cassandraHost: String =  config.getString("lms-cassandra.host")
  val cassandraPort: Int =  config.getInt("lms-cassandra.port")

  val userCacheParallelism: Int = config.getInt("task.usercache.updater.parallelism")

  // constants
  val userSignInTypeKey = "usersignintype"
  val userLoginTypeKey = "userlogintype"
  val stateKey = "state"
  val districtKey = "district"

  // Consumers
  val userCacheConsumer = "user-cache-consumer"

  // Functions
  val userCacheUpdaterFunction = "UserCacheUpdaterFunction"

}

File: data-pipeline/user-cache-updater/src/main/java/org/ekstep/ep/samza/task/UserCacheUpdaterConfig.java
package org.ekstep.ep.samza.task;

import org.apache.samza.config.Config;

import java.util.Arrays;
import java.util.List;

public class UserCacheUpdaterConfig {

    private final String JOB_NAME = "UserCacheUpdater";

    private String userSignInTypeDefault;
    private String userLoginInTypeDefault;
    private List<String> userSelfSignedInTypeList;
    private List<String> userValidatedTypeList;
    private String userSelfSignedKey;
    private String userValidatedKey;
    private String cassandra_db;
    private String cassandra_user_table;
    private String cassandra_location_table;
    private int userStoreDb;

    public UserCacheUpdaterConfig(Config config) {
        this.userSignInTypeDefault = config.get("user.signin.type.default", "Anonymous");
        this.userLoginInTypeDefault = config.get("user.login.type.default", "NA");
        this.userSelfSignedInTypeList = config.getList("user.selfsignedin.typeList", Arrays.asList("google", "self"));
        this.userValidatedTypeList = config.getList("user.validated.typeList", Arrays.asList("sso"));
        this.userSelfSignedKey = config.get("user.self-siginin.key", "Self-Signed-In");
        this.userValidatedKey = config.get("user.valid.key", "Validated");
        this.cassandra_db = config.get("middleware.cassandra.keyspace", "sunbird");
        this.cassandra_user_table = config.get("middleware.cassandra.user_table", "user");
        this.cassandra_location_table = config.get("middleware.cassandra.location_table", "location");
        this.userStoreDb = config.getInt("redis.database.userStore.id", 4);
    }

    public String getUserSignInTypeDefault() { return userSignInTypeDefault; }

    public String getUserLoginInTypeDefault() { return userLoginInTypeDefault; }

    public String getUserSelfSignedKey() { return userSelfSignedKey; }

    public String getUserValidatedKey() { return userValidatedKey; }

    public List<String> getUserSelfSignedInTypeList() { return userSelfSignedInTypeList; }

    public List<String> getUserValidatedTypeList() { return userValidatedTypeList; }

    public String cassandra_db() { return cassandra_db; }

    public String cassandra_user_table() { return cassandra_user_table; }

    public String cassandra_location_table() { return cassandra_location_table; }

    public String jobName() { return JOB_NAME; }

    public int userStoreDb() { return userStoreDb; }

}
